src/java.desktop/share/classes/sun/awt/SunToolkit.java

```diff
@@ -1377,10 +1416,6 @@ public OperationTimedOut() {
         }
     }
 
-    @SuppressWarnings("serial")
-    public static class InfiniteLoop extends RuntimeException {
-    }
-
```

Comment:
I think that the rationale behind this exception as that in case of really stuck loop we could identify it. Why are we throwing it away? Now when we exceed the maximum allowed number of iterations we are assuming that the native event queue synchronization happened while it might not be the case.

Reply:
Yes, that was the initial goal, but currently, it does not work well if some of the components use animation and periodically repainted we hit this exception.